src/main/drivers/rssi_softpwm.c

```diff
@@ -0,0 +1,223 @@
+/* Copyright spell goes here */
+
+//
+// Interrupt based (sloppy) FrSky PWM RSSI
+//
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "platform.h"
+
+#include "build/build_config.h"
+#include "build/atomic.h"
+#include "build/debug.h"
+
+#include "common/maths.h"
+
+#ifdef USE_RSSI_SOFTPWM
+
+#ifndef USE_EXTI
+# error Configuration problem: USE_RSSI_SOFTPWM requires USE_EXTI
+#endif
+
+#include "system.h"
+#include "exti.h"
+#include "io.h"
+#include "gpio.h"
+#include "nvic.h"
+
+#include "config/parameter_group.h"
+#include "config/parameter_group_ids.h"
+
+#include "drivers/rssi_softpwm.h"
+
+// Basic resources
+
+PG_REGISTER_WITH_RESET_TEMPLATE(rssiSoftPwmConfig_t, rssiSoftPwmConfig, PG_RSSI_SOFTPWM_CONFIG, 0);
+
+#ifndef RSSI_SOFTPWM_PIN
+#define RSSI_SOFTPWM_PIN NONE
+#endif
+
+PG_RESET_TEMPLATE(rssiSoftPwmConfig_t, rssiSoftPwmConfig,
+    .ioTag = IO_TAG(RSSI_SOFTPWM_PIN),
+    .min = 0,
+    .minFollow = 0,
+    .monitor = 0,
+);
+
+static const rssiSoftPwmConfig_t *pRspConfig;
+static IO_t rspIO = 0;
+static extiCallbackRec_t rsp_extiCallbackRec;
+
+// Output variable
+static uint16_t rspValue;
+#define RSSI_OUTPUT_MIN     0
+#define RSSI_OUTPUT_MAX  1023
+
+// Measurement Control
+static volatile bool rspInProgress = false;
+
+// Basic measurement by EXTI
+static volatile uint8_t timerState = 0;
+static volatile uint16_t tstamp[3];
+static volatile uint8_t tbits;
+
+// Smoothing
+#define MEDIAN_TAPCOUNT 5
+#define QUICKMEDIANFILTER quickMedianFilter5
+static int32_t dutyFilterArray[MEDIAN_TAPCOUNT];
+static int dutyFilterPos;
+
+// Exception monitoring
+static uint16_t exceptions;
+
+static void rspExtiHandler(extiCallbackRec_t* cb)
+{
+    UNUSED(cb);
+
+    if (!rspInProgress)
+        return;
+
+    switch (timerState) {
+    case 0:
+        tbits = IORead(rspIO);
+        tstamp[0] = micros();
```

Comment:
Ah yes! I actually looked into `micros` for this feature 😉 

`micros` will work also, but `microsISR` is better of course.